### PR TITLE
Dev/ade9000 example

### PR DESCRIPTION
**Pull request by raezt**
## Pull Request Description

Please replace this with a detailed description and motivation of the changes. 
You can tick the checkboxes below with an 'x' between square brackets or just check them after publishing the PR. 
If this PR contains a breaking change, list dependent PRs and try to push all related PRs at the same time.

## PR Type
- [ ] Bug fix (change that fixes an issue)
- [x] New feature (change that adds new functionality)
- [ ] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [x] I have followed the [Coding style guidelines](http://analogdevicesinc.github.io/no-OS/drivers_guide.html#coding-style)
- [x] I have performed a self-review of the changes
- [x] I have commented my code, at least hard-to-understand parts
- [x] I have build all projects affected by the changes in this PR
- [x] I have tested in hardware affected projects, at the relevant boards
- [x] I have signed off all commits from this PR
- [ ] I have updated the documentation (wiki pages, ReadMe etc), if applies

### CODE REVIEW

File: drivers/meter/ade9000/ade9000.c [hunk 1]
```diff
@@ -0,0 +1,528 @@
...
+	/* register addres */
+	uint32_t addr;
+
+	if (!dev)
+		return -ENODEV;
+	if (!reg_data)
+		return -EINVAL;
+
+	/* reset command buffer */
+	for (i = 0; i < 6; i++)
```
Comment (danmois) [inline drivers/meter/ade9000/ade9000.c:71]: Is this needed if already initialized with 0s?

File: drivers/meter/ade9000/ade9000.c [hunk 2]
```diff
@@ -0,0 +1,528 @@
...
+ * @param reg_data - The data read from the register.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ade9000_read(struct ade9000_dev *dev, uint16_t reg_addr, uint32_t *reg_data)
+{
+	int ret;
+	/* index */
+	int i;
+	/* data buffer large enough for 32 bits reg */
+	uint8_t buff[6] = {0};
```
Comment (danmois) [inline drivers/meter/ade9000/ade9000.c:61]: Spaces around 0?

File: drivers/meter/ade9000/ade9000.c [hunk 3]
```diff
@@ -0,0 +1,528 @@
...
+
+	ret = ade9000_read(dev, ADE9000_REG_TEMP_TRIM, &temp);
+	if (ret)
+		return ret;
+
+	gain = no_os_field_get(ADE9000_TEMP_GAIN, temp);
+	offset = no_os_field_get(ADE9000_TEMP_OFFSET, temp);
+
+	/* Temperature (°C) = TEMP_RSLT × (−TEMP_GAIN/65536) + (TEMP_OFFSET/32) */
+	dev->temp_deg = (offset / (1 << 5)) - (temp_raw * ((1000)
```
Comment (danmois) [inline drivers/meter/ade9000/ade9000.c:237]: Are the parentheses around 1000 needed?

File: drivers/meter/ade9000/ade9000.c [hunk 4]
```diff
@@ -0,0 +1,528 @@
...
+
+	ret = ade9000_read(dev, watt_reg, &temp);
+	if (ret)
+		return ret;
+
+	val_int = ADE9000_FS_VOLTAGE * (ADE9000_CURRENT_TR_FCN / 100)
+		  * (ADE9000_FS_VOLTAGE) * (ADE9000_VOLTAGE_TR_FCN / 10);
+	val_int2 = ADE9000_WATT_FS_CODES;
+	val_int = (uint64_t)((uint64_t)temp * (uint64_t)val_int);
+	val_int3 = (uint64_t)((uint64_t)val_int / (uint64_t)val_int2);
```
Comment (danmois) [inline drivers/meter/ade9000/ade9000.c:311]: IS the use of val_int3 required?

File: drivers/meter/ade9000/ade9000.c [hunk 5]
```diff
@@ -0,0 +1,528 @@
...
+	ret = no_os_spi_init(&dev->spi_desc, init_param.spi_init);
+	if (ret)
+		goto error_dev;
+
+	ret = ade9000_update_bits(dev, ADE9000_REG_CONFIG1, ADE9000_SWRST,
+				  no_os_field_prep(ADE9000_SWRST, 1));
+	if (ret)
+		goto error_spi;
+
+	no_os_mdelay(100);
```
Comment (danmois) [inline drivers/meter/ade9000/ade9000.c:411]: Is this 100ms value specified in the datasheet?

File: drivers/meter/ade9000/ade9000.c [hunk 6]
```diff
@@ -0,0 +1,528 @@
...
+	if (ret)
+		return ret;
+	ret = ade9000_write(dev, ADE9000_REG_EGY_TIME, ADE9000_EGY_TIME);
+	if (ret)
+		return ret;
+	ret = ade9000_write(dev, ADE9000_REG_EP_CFG, ADE9000_EP_CFG);
+	if (ret)
+		return ret;
+	/* Enable the DSP */
+	ret = ade9000_write(dev, ADE9000_REG_RUN, ADE9000_RUN_ON);
```
Comment (danmois) [inline drivers/meter/ade9000/ade9000.c:501]: You could return directly here.

File: drivers/meter/ade9000/ade9000.c [hunk 7]
```diff
@@ -0,0 +1,528 @@
...
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ade9000_remove(struct ade9000_dev *dev)
+{
+	int ret;
+
+	if (!dev)
+		return -ENOMEM;
+
+	ret = no_os_spi_remove(dev->spi_desc);
```
Comment (danmois) [inline drivers/meter/ade9000/ade9000.c:521]: Not sure that we are interested in the result here.

File: projects/eval-ade9000/builds.json [hunk 8]
```diff
@@ -0,0 +1,8 @@
+{
+    "maxim": {
+      "ade9000_example": {
+        "flags" : "TARGET=max32690"
+      }
+    }
+  }
+
```
Comment (amiclaus) [inline projects/eval-ade9000/builds.json:8]: nit :) new line

**Review by amiclaus — COMMENTED** [review]
looks good, one nit.

File: drivers/meter/ade9000/ade9000.c (hunk 4 above)
Comment (raezt) [reply drivers/meter/ade9000/ade9000.c:311]: removed val_int2 and val_int3

File: drivers/meter/ade9000/ade9000.c (hunk 5 above)
Comment (raezt) [reply drivers/meter/ade9000/ade9000.c:411]: In the datasheet the amount of time needed for power-on after reset is aprox. 47ms. The value is changed from 100ms to 50ms.

**Comment by raezt** [comment]
- V2 all comments have been addressed .

**Review by danmois — APPROVED** [review]

**Pull request merged**